semtypes/src/main/java/io/ballerina/semtype/RecAtom.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.semtype;
+
+public class RecAtom implements Atom {
+    long index;
```

Comment:
Should be int

Reply:
Java `int` is 32bit, so this should be `long` I think.

Reply:
Only 23 bits are used https://github.com/ballerina-platform/nballerina/blob/fb8c9d506b1c09b267f7aec0ace364f709c969bf/compiler/modules/types/core.bal#L6
so int should work.

Reply:
Ahh this is a different long, I was commenting about UniformTypeCode's long.

Reply:
Number of members in a Java collection cannot exceed what can be represented in an int, so I think it should be int.

Reply:
ahh yes, agreed. Thank you.